kong/init.lua

```diff
@@ -352,15 +352,34 @@ function Kong.access()
   local ctx = ngx.ctx
   core.access.before(ctx)
 
+  ctx.delay_response = true
+
   for plugin, plugin_conf in plugins_iterator(singletons.loaded_plugins, true) do
-    plugin.handler:access(plugin_conf)
+    if not ctx.delayed_response then
+      plugin.handler:access(plugin_conf)
+    end
+  end
+
+  if ctx.delayed_response then
+    return responses.flush_delayed_response(ctx)
   end
 
+  ctx.delay_response = false
+
   core.access.after(ctx)
 end
 
 function Kong.header_filter()
   local ctx = ngx.ctx
+
+  --[[
```

Comment:
I'm having trouble understanding this comment. Assuming the comment code should _not_ run, right? If so can we remove it and just have a human readable comment about the nature of this phase handler?

Reply:
Those were for test purposes only and will be removed before merge - please ignore